libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationEngineProvider.java

```diff
@@ -0,0 +1,68 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import com.mapbox.services.android.navigation.v5.navigation.camera.Camera;
+import com.mapbox.services.android.navigation.v5.navigation.camera.SimpleCamera;
+import com.mapbox.services.android.navigation.v5.offroute.OffRoute;
+import com.mapbox.services.android.navigation.v5.offroute.OffRouteDetector;
+import com.mapbox.services.android.navigation.v5.route.FasterRoute;
+import com.mapbox.services.android.navigation.v5.route.FasterRouteDetector;
+import com.mapbox.services.android.navigation.v5.snap.Snap;
+import com.mapbox.services.android.navigation.v5.snap.SnapToRoute;
+
+class NavigationEngineProvider {
```

Comment:
What about calling this class `NavigationEngineFactory` instead?  I think it'd reflect better its purpose of "providing" different engines. I personally thought that this class was in charge of providing a "navigation" engine when I looked at it in the first place.